media-video/obs-studio: Various updates.

Chiitoo opened the pull request:
Not sure how the QA issue didn't come up last time (maybe something changed in repoman... did not check a look).

Let's see if gentoo-repo-qa-bot yells at me again when doing the Python thingies like so...

Chiitoo commented:
Updated with the bump to be for version 22.0.2 instead of 22.0.1.

a17r commented on media-video/obs-studio/obs-studio-21.1.2.ebuild:
```diff
@@ -6,7 +6,7 @@ EAPI=6
 PYTHON_COMPAT=( python{3_4,3_5,3_6} )
 CMAKE_MIN_VERSION=3.9.6
```
`3.9.6` is current default in `cmake-utils.eclass` so this line could be dropped

Chiitoo replied:
Dropped it shall be.

a17r commented on media-video/obs-studio/obs-studio-22.0.2.ebuild:
```diff
@@ -3,9 +3,10 @@
 
 EAPI=6
 
+PYTHON_COMPAT=( python{3_4,3_5,3_6} )
```
could you check if it may be fine with python3_7 already?

a17r commented on media-video/obs-studio/obs-studio-22.0.2.ebuild:
```diff
@@ -45,17 +46,24 @@ COMMON_DEPEND="
+	nvenc? ( media-video/ffmpeg:=[nvenc] )
 	pulseaudio? ( media-sound/pulseaudio )
+	python? ( ${PYTHON_DEPS} )
+	speexdsp? ( media-libs/speexdsp )
```
Please switch to `USE=speex` here, which is used elsewhere in tree already.

a17r replied:
Yes please. speexdsp and speex are two parts of the same software, and while they are independently packaged for some time now, there is no reason to expect we should follow through with more use flag granularity as well.

Chiitoo commented:
Changes:

- Restricted supported Python versions to 3.6, since that's what upstream seems to require.
- Removed the 'CMAKE_MIN_VERSION'.
- Renamed USE="speexdsp" to "speex".
- Removed the 'use less automagic' patch (upstreamed) from 9999.
- Bumped version bump to 22.0.3 (though it's just a macOS hotfix).

a17r commented:
> * Restricted supported Python versions to 3.6, since that's what upstream seems to require.

Are you sure it does not work with 3.5? Because that change may end up some people with conflicts on upgrade.

Chiitoo commented:
> > * Restricted supported Python versions to 3.6, since that's what upstream seems to require.
> 
> Are you sure it does not work with 3.5? Because that change may end up some people with conflicts on upgrade. I tend to 'keep it if there's no bug report'.

Not at all.  I've heard no reports of it not working, and as such, was indeed wondering which would be the better way forward.

I'll change it back.

Thanks!
